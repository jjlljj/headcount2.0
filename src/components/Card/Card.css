.data-card {
  border: 1px solid black;
  height: 300px;
  margin: 10px;
  width: 240px;
  display: flex;
  flex-direction: column;
  align-items: center;
}

.data-card h5 {
  border-bottom: 1px solid #d2d2d2;
  font-family: Helvetica;
  font-weight: bold;
  font-size: 18px;
  height: 42px;
  margin: 35px 15px 15px;
  text-align: center;
  width: 70%;
}

.data-card ul {

}