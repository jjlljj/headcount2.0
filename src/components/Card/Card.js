import React from 'react';
import propTypes, { func, bool } from 'prop-types';
import './Card.css';

const Card = ({ card, clickHandler, highlight }) => {
  const { location, data } = card;
  const dataKeys = Object.keys(data);

  const classList =  highlight ? "highlight data-card" : "data-card";

  const renderedData = dataKeys.map( dp => {
    const dataStyle = data[dp] <= 0.5 ? 'red' : 'blue'; 
    return (
      <li key={dp} >
        {dp} : <span className={dataStyle}>{data[dp]}</span>
      </li>
    );
  });

  return (

    <div  
      className={ classList }
      onClick={() => clickHandler(location)}>
      <h5>{location}</h5>
      <ul>
        { renderedData } 
      </ul>
    </div>
  );
};

Card.propTypes = {
  card: propTypes.oneOfType([propTypes.string, propTypes.object]),
  clickHandler: func,
  highlight: bool
};

export default Card;


