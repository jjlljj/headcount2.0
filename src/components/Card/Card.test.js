/* eslint-disable */ 

import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme'
import Card from './Card';

describe('Card', () => {

  it('should exist', () => {
    const mockData = { location: 'Denver, Co', data: { 2004: 0.007, 2005: 1.112, 200: 7.110 } }
    const renderedComponent = shallow(<Card card={mockData}/>)

    expect(renderedComponent).toBeDefined()
  })

  it('should should match its snapshot', () => {
    const mockData = { location: 'Denver, Co', data: { 2004: 0.007, 2005: 1.112, 200: 7.110 } }
    const renderedComponent = shallow(<Card card={mockData}/>)

    expect(renderedComponent).toMatchSnapshot();
  })

  it('should fire the clickHandler when it is clicked', () => {
    const mockData = { location: 'Denver, Co', data: { 2004: 0.007, 2005: 1.112, 200: 7.110 } }
    const mockClickHandler = jest.fn()
    const renderedComponent = shallow(<Card card={mockData} clickHandler={mockClickHandler}/>)

    renderedComponent.find('div').simulate('click')

    expect(mockClickHandler).toHaveBeenCalled();
  })
  
});