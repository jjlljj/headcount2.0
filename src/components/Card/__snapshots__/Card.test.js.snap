// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should should match its snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card
    card={
        Object {
            "data": Object {
              "200": 7.11,
              "2004": 0.007,
              "2005": 1.112,
            },
            "location": "Denver, Co",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h5>
          Denver, Co
</h5>,
        <ul>
          <li>
                    <i
                              aria-hidden="true"
                              className="fa fa-check"
                    />
                    200
                     : 
                    <span
                              className=" blue"
                    >
                              7.11
                    </span>
          </li>
          <li>
                    <i
                              aria-hidden="true"
                              className=""
                    />
                    2004
                     : 
                    <span
                              className="red"
                    >
                              0.007
                    </span>
          </li>
          <li>
                    <i
                              aria-hidden="true"
                              className="fa fa-check"
                    />
                    2005
                     : 
                    <span
                              className=" blue"
                    >
                              1.112
                    </span>
          </li>
</ul>,
      ],
      "className": "data-card",
      "onClick": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Denver, Co",
        },
        "ref": null,
        "rendered": "Denver, Co",
        "type": "h5",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <li>
              <i
                            aria-hidden="true"
                            className="fa fa-check"
              />
              200
               : 
              <span
                            className=" blue"
              >
                            7.11
              </span>
</li>,
            <li>
              <i
                            aria-hidden="true"
                            className=""
              />
              2004
               : 
              <span
                            className="red"
              >
                            0.007
              </span>
</li>,
            <li>
              <i
                            aria-hidden="true"
                            className="fa fa-check"
              />
              2005
               : 
              <span
                            className=" blue"
              >
                            1.112
              </span>
</li>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "200",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <i
                  aria-hidden="true"
                  className="fa fa-check"
/>,
                "200",
                " : ",
                <span
                  className=" blue"
>
                  7.11
</span>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "aria-hidden": "true",
                  "className": "fa fa-check",
                },
                "ref": null,
                "rendered": null,
                "type": "i",
              },
              "200",
              " : ",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": 7.11,
                  "className": " blue",
                },
                "ref": null,
                "rendered": 7.11,
                "type": "span",
              },
            ],
            "type": "li",
          },
          Object {
            "instance": null,
            "key": "2004",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <i
                  aria-hidden="true"
                  className=""
/>,
                "2004",
                " : ",
                <span
                  className="red"
>
                  0.007
</span>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "aria-hidden": "true",
                  "className": "",
                },
                "ref": null,
                "rendered": null,
                "type": "i",
              },
              "2004",
              " : ",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": 0.007,
                  "className": "red",
                },
                "ref": null,
                "rendered": 0.007,
                "type": "span",
              },
            ],
            "type": "li",
          },
          Object {
            "instance": null,
            "key": "2005",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <i
                  aria-hidden="true"
                  className="fa fa-check"
/>,
                "2005",
                " : ",
                <span
                  className=" blue"
>
                  1.112
</span>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "aria-hidden": "true",
                  "className": "fa fa-check",
                },
                "ref": null,
                "rendered": null,
                "type": "i",
              },
              "2005",
              " : ",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": 1.112,
                  "className": " blue",
                },
                "ref": null,
                "rendered": 1.112,
                "type": "span",
              },
            ],
            "type": "li",
          },
        ],
        "type": "ul",
      },
    ],
    "type": "article",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h5>
            Denver, Co
</h5>,
          <ul>
            <li>
                        <i
                                    aria-hidden="true"
                                    className="fa fa-check"
                        />
                        200
                         : 
                        <span
                                    className=" blue"
                        >
                                    7.11
                        </span>
            </li>
            <li>
                        <i
                                    aria-hidden="true"
                                    className=""
                        />
                        2004
                         : 
                        <span
                                    className="red"
                        >
                                    0.007
                        </span>
            </li>
            <li>
                        <i
                                    aria-hidden="true"
                                    className="fa fa-check"
                        />
                        2005
                         : 
                        <span
                                    className=" blue"
                        >
                                    1.112
                        </span>
            </li>
</ul>,
        ],
        "className": "data-card",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Denver, Co",
          },
          "ref": null,
          "rendered": "Denver, Co",
          "type": "h5",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <li>
                <i
                                aria-hidden="true"
                                className="fa fa-check"
                />
                200
                 : 
                <span
                                className=" blue"
                >
                                7.11
                </span>
</li>,
              <li>
                <i
                                aria-hidden="true"
                                className=""
                />
                2004
                 : 
                <span
                                className="red"
                >
                                0.007
                </span>
</li>,
              <li>
                <i
                                aria-hidden="true"
                                className="fa fa-check"
                />
                2005
                 : 
                <span
                                className=" blue"
                >
                                1.112
                </span>
</li>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "200",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <i
                    aria-hidden="true"
                    className="fa fa-check"
/>,
                  "200",
                  " : ",
                  <span
                    className=" blue"
>
                    7.11
</span>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "aria-hidden": "true",
                    "className": "fa fa-check",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "i",
                },
                "200",
                " : ",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": 7.11,
                    "className": " blue",
                  },
                  "ref": null,
                  "rendered": 7.11,
                  "type": "span",
                },
              ],
              "type": "li",
            },
            Object {
              "instance": null,
              "key": "2004",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <i
                    aria-hidden="true"
                    className=""
/>,
                  "2004",
                  " : ",
                  <span
                    className="red"
>
                    0.007
</span>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "aria-hidden": "true",
                    "className": "",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "i",
                },
                "2004",
                " : ",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": 0.007,
                    "className": "red",
                  },
                  "ref": null,
                  "rendered": 0.007,
                  "type": "span",
                },
              ],
              "type": "li",
            },
            Object {
              "instance": null,
              "key": "2005",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <i
                    aria-hidden="true"
                    className="fa fa-check"
/>,
                  "2005",
                  " : ",
                  <span
                    className=" blue"
>
                    1.112
</span>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "aria-hidden": "true",
                    "className": "fa fa-check",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "i",
                },
                "2005",
                " : ",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": 1.112,
                    "className": " blue",
                  },
                  "ref": null,
                  "rendered": 1.112,
                  "type": "span",
                },
              ],
              "type": "li",
            },
          ],
          "type": "ul",
        },
      ],
      "type": "article",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
